source/simulator/helper_functions.cc

```diff
@@ -1526,8 +1526,15 @@ namespace aspect
 // explicit instantiation of the functions we implement in this file
 namespace aspect
 {
+#ifndef aspect_simulator_defined
+#define MAYBE_ADVECTION_FIELD(dim) \
+  template struct Simulator<dim>::AdvectionField;
```

Comment:
Yes, this I would need some help with (template problems). 
When I do a unity build with an explicit instantiation of the Simulator class and this line in one translation unit, I get the following error:
```
In file included from /scratch/gassmoel/aspect-debug/cotire/aspect_CXX_unity.cxx:2118:0:
/raid/gassmoel/software/aspect/source/simulator/helper_functions.cc:1537:35: error: duplicate explicit instantiation of ‘struct aspect::Simulator<2>::AdvectionField’ [-fpermissive]
   template struct Simulator<dim>::AdvectionField; \
                                   ^
/raid/gassmoel/software/aspect/include/aspect/global.h:384:3: note: in expansion of macro ‘INSTANTIATE’
   INSTANTIATIONS(2) \
   ^
/raid/gassmoel/software/aspect/source/simulator/helper_functions.cc:1555:3: note: in expansion of macro ‘ASPECT_INSTANTIATE’
   ASPECT_INSTANTIATE(INSTANTIATE)
   ^
/raid/gassmoel/software/aspect/source/simulator/helper_functions.cc:1537:35: error: duplicate explicit instantiation of ‘struct aspect::Simulator<3>::AdvectionField’ [-fpermissive]
   template struct Simulator<dim>::AdvectionField; \
                                   ^
/raid/gassmoel/software/aspect/include/aspect/global.h:385:3: note: in expansion of macro ‘INSTANTIATE’
   INSTANTIATIONS(3)
   ^
/raid/gassmoel/software/aspect/source/simulator/helper_functions.cc:1555:3: note: in expansion of macro ‘ASPECT_INSTANTIATE’
   ASPECT_INSTANTIATE(INSTANTIATE)
   ^
make[3]: *** [CMakeFiles/aspect_unity.dir/cotire/aspect_CXX_unity.cxx.o] Error 1
make[2]: *** [CMakeFiles/aspect_unity.dir/all] Error 2
make[1]: *** [CMakeFiles/aspect_unity.dir/rule] Error 2
```